doc/BRANCHING-AND-RELEASE.adoc

```diff
@@ -0,0 +1,126 @@
+
+There are different kinds of node releases: a node release can be a non-hard fork release, an intra-era hard fork release (i.e., `PV` will be incremented, but not `ERA`), or a next-era hard fork release.
+In the following, assume `nY` to be a next-era hard fork release, which is the most complex case.
+
+- `main` depends on node `nX`, and `next-node` tracks node master.
+* `next-node` should regularly updates its node commit. Ideally we want `next-node` to depend on the tip of node master, but we are constrained by cardano-wallet, which is a PAB dependency and depends on the node.
+If cardano-wallet does not build with the tip of node master, we have to either use an older node commit, or temporarily live with a PAB that doesn't build.
+* A Plutus Tools PR can be opened against either `main` or `next-node`, determined by whether or not it depends on new upstream features or functionalities not available in `nX`.
+* In the `next-node` branch, we can start developing features and functionalities for the next ledger era, `ERA+1`.
+But (1) we should make it clear in the API/UI which features and functionalities only work in `ERA+1`, i.e., not yet usable;
+(2) we should not remove any features for the current ledger era, `ERA`.
+The reasons for these will become apparent later.
+- New Plutus Tools releases can be done from `main` as scheduled or as needed.
+* A release at this point is compatible with `nX`, protocol version `PV` and ledger era `ERA`.
+See "Release Process" for how to make releases.
+- Cardano node starts the release process for version `nY` by creating a release branch for `nY`.
+* The `next-node` branch should now track the release branch for `nY`, instead of master.
+* Users testing the Cardano node release candidates, `nY-rc<x>`, on devnets and testnets, can build Plutus Tools from the `next-node` branch, which would be compatible with `nY-rc<x>`.
+- Cardano node version `nY` is published.
+* (Optional) we release a new Plutus Tools version `tX'` from `main`.
+** This is optional because it is only useful if `main` has new features worth releasing for users still on node `nX`.
+Most users should start upgrading to node `nY` at this point.
+** If we do so, `tX'` would be the last Plutus Tools release compatible with node version `nX`.
+- We merge `next-node` into `main`, and release a new Plutus Tools version `tY`.
+* `tY` is the first Plutus Tools version compatible with node version `nY`.
+* The hard fork hasn't happened at this point, and we are still on ledger era `ERA`.
+Therefore `tY` must remain compatible with `ERA`.
+Specifically, `tY` may contain features and functionalities for `ERA+1` (such as the ability to build `ERA+1` transactions), but since they are not yet usable, it is helpful to make it clear which features and functionalities only work in `ERA+1`.
+Also, this is why the `next-node` branch should not remove any features for ledger era `ERA`, until we are in ledger era `ERA+1`.
```

Comment:
I understand the rationale, but do we need this stage?
It may slow down the deployment of a fully compatible `nY` version (if we consider that `nY` introduce breaking changes from `nX`) and add extra work to remove deprecated features afterwards.